src/tests/IntegrationTest.test.jsx

```diff
@@ -0,0 +1,82 @@
+/* eslint-disable import/no-extraneous-dependencies */
+import {
+  test,
+  expect,
+  afterEach,
+  describe,
+  vi,
+} from 'vitest';
+import {
+  cleanup,
+  render,
+  screen,
+} from '@testing-library/react';
+import { Provider } from 'react-redux';
+import { BrowserRouter, useParams } from 'react-router-dom';
+import HomePage from '../pages/HomePage';
+import Header from '../components/Header';
+import App from '../components/App';
+import store from '../redux/configureStore';
+
+afterEach(cleanup);
+
+describe('Test App component', () => {
```

Comment:
- [ ] I can see you have wrote perfect unit tests. However, I was not able to find any integration tests in your code. It would be nice to have test your functional side of your code example: fetching data... Kindly add integration tests in it.